### PR TITLE
Doc: add page for geoc conversion (fixes #1092)

### LINKED ISSUE #1092: Undocumented +proj=geoc
Unless I'm mistaken the +proj=geoc for geodetic latitude to geocentric latitude is not documented.

### PR DESCRIPTION
**Pull request by rouault**
And a fix for merc, and some improvements for cart

### CODE REVIEW

File: docs/source/operations/conversions/geoc.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+in the context of non-Earth bodies) is the angle between the equatorial plane
+and the normal (vertical) to the ellipsoid surface at the considered point.
+The geodetic latitude is what is normally used everywhere in PROJ when angular
+coordinates are expected or produced.
+
+The geocentric latitude (also called planetocentric latitude in the context of
+non-Earth bodies) is the angle between the equatorial plane and a line joining
+the body centre to the considered point.
+
+Note: this conversion must be distinguished fom the :ref:`cart`
```
Comment (kbevers) [inline docs/source/operations/conversions/geoc.rst]: Use `.. note::` to display a proper note textbox

**Review by kbevers — COMMENTED** [review]
Looks good to me.

A possible improvement is a figure that explains the difference between geodetic and geocentric latitude (can we steal one from Wikipedia?).

File: docs/source/operations/conversions/geoc.rst (hunk 1 above)
Comment (rouault) [reply docs/source/operations/conversions/geoc.rst]: fixed

**Comment by rouault** [comment]
I've added my own sketch + some improvements in travis/build_docs.sh to make it error on exit, and to be able to easily run it locally more easily

**Pull request merged**